src/main/java/org/openrewrite/java/logging/slf4j/MatchIsLogLevelEnabledWithLogStatements.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.logging.slf4j;
+
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Statement;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.nullsFirst;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.BinaryOperator.maxBy;
+
+public class MatchIsLogLevelEnabledWithLogStatements extends Recipe {
+
+    private static final MethodMatcher ENABLED_MATCHER = new MethodMatcher("org.slf4j.Logger is*Enabled()");
+    private static final MethodMatcher LOG_MATCHER = new MethodMatcher("org.slf4j.Logger *(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "Match `if (is*Enabled())` with logging statements";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Change any `if (is*Enabled())` statements that do not match the maximum log level used in the `then` " +
+                "part to use the matching `is*Enabled()` method for that log level. " +
+                "This ensures that the logging condition is consistent with the actual logging statements.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesMethod<>(ENABLED_MATCHER),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.If visitIf(J.If iff, ExecutionContext ctx) {
+                        J.If if_ = super.visitIf(iff, ctx);
+                        if (if_.getIfCondition().getTree() instanceof J.MethodInvocation && if_.getElsePart() == null) {
+                            J.MethodInvocation enabledMethodInvocation = (J.MethodInvocation) if_.getIfCondition().getTree();
```

Comment:
Though I understand the name of the variable, I think it's a confusing name because it looks like (if you read quickly) you are using a "enabled method invocation" here, meaning a method invocation that is "on" somehow. I think I would just just the `mi` shorthand here.